emperor/support_files/js/draw.js

```diff
@@ -295,11 +299,80 @@ define(['underscore', 'three', 'jquery'], function(_, THREE, $) {
     // we increase the number of points to have a smoother transition on
     // edges i. e. where the trajectory changes the direction it is going
-    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * 3, radius,
-                                          10, false);
+    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * NUM_TUBE_SEGMENTS, radius,
+                                          NUM_TUBE_CROSS_SECTION_POINTS, false);
+
+    return new THREE.Mesh(lineGeometry, material);
+  }
+  
+  function drawTrajectoryLineDynamic(trajectory, currentFrame, color, radius) {
+    // based on the example described in:
+    // https://github.com/mrdoob/three.js/wiki/Drawing-lines
+    var material, points = [], lineGeometry, limit = 0, path;
+
+    _trajectory = trajectory.representativeInterpolatedCoordinatesAtIndex(currentFrame);
+    if (_trajectory === null || _trajectory.length == 0)
+      return null;
+
+    material = new THREE.MeshPhongMaterial({color: color});
+    material.matrixAutoUpdate = true;
+    material.transparent = false;
+
+    for (var index = 0; index < _trajectory.length; index++) {
+      points.push(new THREE.Vector3(_trajectory[index].x,
+                  _trajectory[index].y, _trajectory[index].z));
+    }
+
+    path = new THREE.EmperorTrajectory(points);
+    
+    // the line will contain the two vertices and the described material
+    // we increase the number of points to have a smoother transition on
+    // edges i. e. where the trajectory changes the direction it is going
+    lineGeometry = new THREE.TubeGeometry(path, (points.length - 1) * NUM_TUBE_SEGMENTS, radius,
+                                          NUM_TUBE_CROSS_SECTION_POINTS, false);
 
     return new THREE.Mesh(lineGeometry, material);
   }
+    
+  function drawTrajectoryLineStatic(trajectory, color, radius) {
+    var _trajectory = trajectory.coordinates;
+    
+    var material = new THREE.MeshPhongMaterial({color: color});
+    material.matrixAutoUpdate = true;
+    material.transparent = false;
+    
+    var allPoints = [];
+    for (var index = 0; index < _trajectory.length; index++) {
+      allPoints.push(new THREE.Vector3(_trajectory[index].x,
+                  _trajectory[index].y, _trajectory[index].z));
+    }
+
+    var path = new THREE.EmperorTrajectory(allPoints);
+    
+    //Tubes are straight segments, but adding vertices along them might change lighting effects
+    //under certain models and lighting conditions.
+    var tubeBufferGeom = new THREE.TubeBufferGeometry(path, (allPoints.length - 1) * NUM_TUBE_SEGMENTS, radius, NUM_TUBE_CROSS_SECTION_POINTS, false)
+
+    return new THREE.Mesh(tubeBufferGeom, material);
+  }
+  
+  function updateStaticTrajectoryDrawRange(trajectory, currentFrame, threeMesh)
+  {
+    //Blah, reverse engineering the number of points in a THREE tube is not fun, and may be implementation/version dependent.
```

Comment:
Blah?

Reply:
Is there a preferred comment to mark functions we think most likely to break?  
I personally tend to run with TODO FIXME HACK.  I'll remove the Blah.

Reply:
FIXME should be good. Let's ensure we exercise this enough that we are confident this won't break easily 👍

Reply:
Duplicate of ElDeveloper's comments on draw.js above.  Collapsing conversation.  Target should be to make the updateStaticTrajectoryDrawRange function as robust as possible.